Can't compile generated C

### Describe the bug

 I would like to write a shared library in V and compile it in C to then use it in a C++ program.
When I try to compile the C++ program, I have errors that are found in the C file that was generated by V

### Reproduction Steps

Here is the V file:
`module test_print

@[export: 'get_name']
fn get_name() &char {
    return c'Test'
}`

I used the following command to compile it to name.c: `v -shared -gc none -o name.c name.v`.

Here is the cpp file:
```
#include <iostream>

extern "C" {
    char* get_name(void);
}

int main(int argc, char *argv[])
{
    std::cout << get_name() << '\n';
}
```

And here is the command I use to compile the whole thing: `cl main.cpp name.c /Fe:main.exe`.

And finally the error (sorry it's in french).
```Compilateur d'optimisation Microsoft (R) C/C++ version 19.41.34123 pour x64
Copyright (C) Microsoft Corporation. Tous droits réservés.

main.cpp
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\ostream(787): warning C4530: Gestionnaire d'exceptions C++ utilisé, mais les sémantiques de déroulement n'ont pas été activées. Spécifiez /EHsc
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\ostream(787): note: le contexte d’instanciation du modèle (le plus ancien) est
main.cpp(9): note: voir la référence à l'instanciation de la fonction modèle 'std::basic_ostream<char,std::char_traits<char>> &std::operator <<<std::char_traits<char>>(std::basic_ostream<char,std::char_traits<char>> &,const char *)' en cours de compilation
Génération de code en cours...
Compilation en cours...
name.c
name.c(1724): error C2143: erreur de syntaxe : absence de ')' avant '('
name.c(1724): error C2059: erreur de syntaxe : ')'
name.c(1724): error C2059: erreur de syntaxe : ')'
name.c(1724): error C2146: erreur de syntaxe : absence de ')' avant l'identificateur 'unhandled_exception_handler'
name.c(1724): error C2091: fonction retournée par la fonction
name.c(1724): error C2061: erreur de syntaxe : identificateur 'unhandled_exception_handler'
name.c(1724): error C2059: erreur de syntaxe : ';'
name.c(1724): error C2059: erreur de syntaxe : '<parameter-list>'
name.c(7255): warning C4028: paramètre formel 1 différent de la déclaration
name.c(7623): warning C4477: 'fprintf' : la chaîne de format '%ld' nécessite un argument de type 'long', mais l'argument variadique 1 est de type 'isize'
name.c(7623): note: utilisez '%td' dans la chaîne de format
name.c(8109): error C2143: erreur de syntaxe : absence de ')' avant '('
name.c(8109): error C2059: erreur de syntaxe : ')'
name.c(8109): error C2059: erreur de syntaxe : ')'
name.c(8109): error C2146: erreur de syntaxe : absence de ')' avant l'identificateur 'unhandled_exception_handler'
name.c(8109): error C2091: fonction retournée par la fonction
name.c(8109): error C2061: erreur de syntaxe : identificateur 'unhandled_exception_handler'
name.c(8109): error C2059: erreur de syntaxe : ';'
name.c(8109): error C2059: erreur de syntaxe : '<parameter-list>'
name.c(8125): error C2065: 'unhandled_exception_handler' : identificateur non déclaré
name.c(8125): warning C4312: 'cast de type' : la conversion de 'int' en 'voidptr' d'une taille supérieure```

### Expected Behavior

I should be able to compile the C++ file.

### Current Behavior

It seems that msvc is not able to compile the file generated c file because I have compilation errors.

### Possible Solution

I works with the option `-cc g++`: `v -shared -gc none -cc g++ -o name.c name.v`

### Additional Information/Context

_No response_

### V version

V 0.4.8 6c94c24

### Environment details (OS name and version, etc.)

Windows 11
Visual Studio 17 2022

--- a/vlib/v/gen/c/cmain.v
+++ b/vlib/v/gen/c/cmain.v
@@ -134,7 +134,9 @@ fn (mut g Gen) gen_c_main_function_only_header() {
 fn (mut g Gen) gen_c_main_function_header() {
 	g.gen_c_main_function_only_header()
 	g.gen_c_main_trace_calls_hook()
-	g.writeln2('\tg_main_argc = ___argc;', '\tg_main_argv = ___argv;')
+	if !g.pref.no_builtin {
+		g.writeln2('\tg_main_argc = ___argc;', '\tg_main_argv = ___argv;')
+	}
 	if g.nr_closures > 0 {
 		g.writeln('\t__closure_init(); // main()')
 	}
@@ -158,7 +160,9 @@ fn (mut g Gen) gen_c_main_header() {
 		}
 		g.writeln('#endif')
 	}
-	g.writeln('\t_vinit(___argc, (voidptr)___argv);')
+	if !g.pref.no_builtin {
+		g.writeln('\t_vinit(___argc, (voidptr)___argv);')
+	}
 	g.gen_c_main_profile_hook()
 	if g.pref.is_livemain {
 		g.generate_hotcode_reloading_main_caller()
@@ -166,7 +170,9 @@ fn (mut g Gen) gen_c_main_header() {
 }
 
 pub fn (mut g Gen) gen_c_main_footer() {
-	g.writeln('\t_vcleanup();')
+	if !g.pref.no_builtin {
+		g.writeln('\t_vcleanup();')
+	}
 	g.writeln2('\treturn 0;', '}')
 }
 
@@ -208,9 +214,9 @@ sapp_desc sokol_main(int argc, char* argv[]) {
 		}
 		g.writeln('#endif')
 	}
-	g.writeln('\t_vinit(argc, (voidptr)argv);
-	')
-
+	if !g.pref.no_builtin {
+		g.writeln('\t_vinit(argc, (voidptr)argv);')
+	}
 	g.gen_c_main_profile_hook()
 	g.writeln('\tmain__main();')
 	if g.is_autofree {
@@ -290,7 +296,9 @@ pub fn (mut g Gen) gen_c_main_for_tests() {
 		}
 		g.writeln('#endif')
 	}
-	g.writeln('\t_vinit(___argc, (voidptr)___argv);')
+	if !g.pref.no_builtin {
+		g.writeln('\t_vinit(___argc, (voidptr)___argv);')
+	}
 	g.writeln('\tmain__vtest_init();')
 	g.gen_c_main_profile_hook()
 

